hack/acknowledge_critical_fix_repos_single_repo.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+"""This script handles turning on acknowledge-critical-fixes-only for a
+single branch on a single repository."""
+import copy
+import sys
+import os
+import yaml
+
+def split_out_branch(repo, branch_to_split, revert=False):
+    """Splits out branch from a prow config"""
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    yaml_file = os.path.join(script_dir,
+                             "../core-services/prow/02_config", repo, "_prowconfig.yaml")
+
+    # Check if the file exists
+    if not os.path.isfile(yaml_file):
+        print(f"YAML file not found: {yaml_file}")
+        sys.exit(1)
+
+    with open(yaml_file, 'r', encoding='utf-8') as file:
+        config = yaml.safe_load(file)
+
+    if revert:
+        new_queries = []
+        for query in config['tide']['queries']:
+            if 'includedBranches' in query and branch_to_split in query['includedBranches']:
+                if len(query['includedBranches']) == 1:
+                    continue  # Remove this query as it's the split-out branch
+                query['includedBranches'].remove(branch_to_split)
+            new_queries.append(query)
+
+        # Add the branch back to the original query
+        new_queries[0]['includedBranches'].append(branch_to_split)
+        config['tide']['queries'] = new_queries
+    else:
+        queries = config['tide']['queries'][0] # Don't assume it's the only one
+        new_query = copy.deepcopy(queries)
+
+        if branch_to_split in queries['includedBranches']:
+            queries['includedBranches'].remove(branch_to_split)
+            new_query['includedBranches'] = [branch_to_split]
+            new_query['labels'].append('acknowledge-critical-fixes-only')
+            config['tide']['queries'].append(new_query)
+        else:
+            print(f"Branch '{branch_to_split}' not found in the includedBranches.")
+            return
```

Comment:
i'd have defined a function for each case and passed that function into `split_out_branch` just to be tidy, but it's hardly important in a script this size